### ISSUE TITLE
Strange error when calling len() on a class without __len__ and then using the result for division

### ISSUE THREAD

**Issue by thomkeh** [python/mypy#12897]
**Bug Report**

Consider this code:

```python
class A: ...

b = 3.0
c = b / len(A())
reveal_type(c)
```

mypy gives this result:

```
division.py:4: error: Unsupported operand types for / ("float" and "int")
division.py:4: error: Argument 1 to "len" has incompatible type "A"; expected "Sized"
division.py:5: note: Revealed type is "Any"
Found 2 errors in 1 file (checked 1 source file)
```

The second error is expected, but the first one (“Unsupported operand types for / ("float" and "int").”) is very confusing, because clearly division is defined for `float` and `int`.

If `A` has `__len__`, everything is fine:

```python
class A:
    def __len__(self) -> int: return 0

b = 3.0
c = b / len(A())
reveal_type(c)
```
mypy output:
```
division.py:6: note: Revealed type is "builtins.float"
Success: no issues found in 1 source file
```

**Your Environment**

- Mypy version used: 0.960
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files): none
- Python version used: 3.10
- Operating system and version: Ubuntu 22.04
